### PR TITLE
stochas: init at 1.3.4

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/stochas/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+stdenv.mkDerivation rec {
+  pname = "stochas";
+  version = "1.3.4";
+
+  src = fetchFromGitHub {
+    owner = "surge-synthesizer";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0b26mbj727dnygavz4kihnhmnnvwsr9l145w6kydq7bd7nwiw7lq";
+    leaveDotGit = true; # for SURGE_VERSION
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/stochas/default.nix:12]: Can we set this with ``cmake -B build -DSTOCHAS_VERSION=xxx``? https://github.com/surge-synthesizer/stochas/blob/main/CMakeLists.txt#L6
``

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107577` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>stochas</li>
  </ul>
</details>

File: pkgs/applications/audio/stochas/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/applications/audio/stochas/default.nix:12]: Please elaborate, I'm not following.

**Comment by magnetophon** [comment]
What error are you getting?

File: pkgs/applications/audio/stochas/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/stochas/default.nix:12]: It is better for reproducibility to remove the .git directory. We can set the version maybe with this flag or through patching.

**Comment by SuperSandro2000** [comment]
```
hash mismatch in fixed-output derivation '/nix/store/x5m3hfkynccdw5hsgc7kd7s01l8ksqm4-source':                                                                                                wanted: sha256:0b26mbj727dnygavz4kihnhmnnvwsr9l145w6kydq7bd7nwiw7lq                                                                                                                         got:    sha256:0qq66zbiiyq0baibwabim8y1i5m945m41gr3lrfw5hcx56a83flh                                                                                                                       cannot build derivation '/nix/store/m2n8a56bnz8kh15najjsdij0fajxfz8v-stochas-1.3.4.drv': 1 dependencies couldn't be built                                                                   cannot build derivation '/nix/store/xfap56i2smhmsbl03s876gral111z9vh-env.drv': 1 dependencies couldn't be built
```

File: pkgs/applications/audio/stochas/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/applications/audio/stochas/default.nix]: Looks like I made a mistake.
When I remove ``leaveDotGit = true;`` and change the hash, I get back the original hash from the build.
When I change it back, there is no build and the original pkg get's installed.

**Comment by magnetophon** [comment]
Hmm, cannot reproduce that here.
What can I do to troubleshoot?

**Comment by SuperSandro2000** [comment]
> Hmm, cannot reproduce that here.
> What can I do to troubleshoot?

change the hash and see if you get the same.

**Comment by magnetophon** [comment]
I did, and the hash comes back the same.

File: pkgs/applications/audio/stochas/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  installPhase = ''
+    mkdir -p $out/lib/vst3
+    cp -r stochas_artefacts/Release/VST3/Stochas.vst3 $out/lib/vst3
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Probabilistic polyrhythmic sequencer plugin";
+    homepage = "https://stochas.org/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/stochas/default.nix:29]: `gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107577` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stochas</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107577` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>stochas</li>
  </ul>
</details>

**Review by SuperSandro2000 — COMMENTED** [review]
Not sure why the hash was mismatched.

File: pkgs/applications/audio/stochas/default.nix (hunk 2 above)
Comment (magnetophon) [reply pkgs/applications/audio/stochas/default.nix]: Done.

**Pull request merged**